### PR TITLE
Chapter 8: Basic login

### PR DESCRIPTION
**Pull request by tnhs97**
image

### CODE REVIEW

**Comment by tnhs97** [comment]
ready

File: config/routes.rb [hunk 1]
```diff
@@ -6,6 +6,9 @@
   get "/contact", to: "static_pages#contact"
   get "/signup", to: "users#new"
   post "/signup", to: "users#create"
+  get '/login', to: 'sessions#new'
+  post '/login', to: 'sessions#create'
+  delete '/logout', to: 'sessions#destroy'
```
Comment (phamvanbk11) [inline config/routes.rb]: Nháy "

File: app/controllers/sessions_controller.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
+class SessionsController < ApplicationController
+  def new; end
+
+  def create
+    user = User.find_by(email: params[:session][:email].downcase)
+    if user&.authenticate(params[:session][:password])
+      log_in user
+      redirect_to user
+    else
+      flash.now[:danger] = "Invalid email/password combination"
```
Comment (phamvanbk11) [inline app/controllers/sessions_controller.rb]: I18n nhé em

File: app/views/layouts/_header.html.erb [hunk 3]
```diff
@@ -5,7 +5,24 @@
       <ul class="nav navbar-nav navbar-right">
         <li><%= link_to t("home"), root_path %></li>
         <li><%= link_to t("help"), help_path %></li>
-        <li><%= link_to t("log_in"), "#" %></li>
+        <% if logged_in? %>
+          <li><%= link_to "Users", '#' %></li>
```
Comment (phamvanbk11) [inline app/views/layouts/_header.html.erb]: I18n nhé

**Comment by phamvanbk11** [comment]
commented

**Comment by tnhs97** [comment]
ready

**Comment by phamvanbk11** [comment]
merged

**Pull request merged**